DigitalLearningSolutions.Web/Controllers/MyAccountController.cs

```diff
@@ -49,23 +56,83 @@ public IActionResult EditDetails()
         }
 
         [HttpPost]
-        public IActionResult EditDetails(EditDetailsViewModel model)
+        public IActionResult EditDetails(EditDetailsViewModel model, string action)
+        {
+            return action switch
+            {
+                "save" => EditDetailsPostSave(model),
+                "previewImage" => EditDetailsPostPreviewImage(model),
+                "removeImage" => EditDetailsPostRemoveImage(model),
+                _ => View(model),
+            };
+        }
+
+        private IActionResult EditDetailsPostSave(EditDetailsViewModel model)
         {
             if (!ModelState.IsValid)
             {
                 return View(model);
             }
 
+            if (model.ProfilePicture != null)
+            {
+                ModelState.AddModelError("ProfilePicture", "Preview your new profile picture before saving");
+                return View(model);
+            }
+
             var userAdminId = User.GetAdminId();
             var userDelegateId = User.GetNullableCandidateId();
 
-            if (!userService.TryUpdateUserAccountDetails(userAdminId, userDelegateId, model.Password,  model.FirstName, model.LastName, model.Email))
+            var profileImageToSave = model.HasProfileImageBeenRemoved ? null : model.NewProfileImage ?? model.CurrentProfileImage;
+
+            if (!userService.TryUpdateUserAccountDetails(userAdminId, userDelegateId, model.Password,  model.FirstName, model.LastName, model.Email, profileImageToSave))
             {
                 ModelState.AddModelError("Password", "The password you have entered is incorrect.");
                 return View(model);
             }
 
             return RedirectToAction("Index");
         }
+
+        private IActionResult EditDetailsPostPreviewImage(EditDetailsViewModel model)
+        {
+            // We don't want to display validation errors on other fields in this case
+            foreach (var key in ModelState.Keys.Where(k => k != "ProfilePicture"))
+            {
+                ModelState[key].Errors.Clear();
+                ModelState[key].ValidationState = ModelValidationState.Valid;
+            }
+
+            if (!ModelState.IsValid)
+            {
+                return View(model);
+            }
+
+            // Remove the old new profile image value from the ModelState
+            ModelState.Remove("NewProfileImage");

```

Comment:
Due to the way all the modelstate stuff works it needs to be removed before setting in the post otherwise it will keep its value if you go through this method a second time, e.g. if the crop was bad so you tried a different image, it'll still contain the old image and won't overwrite properly. I don't really understand why it behaves like this, but that's what I found. I can look to see if I can get one of these alternatives to work correctly.